### ISSUE TITLE
Build docker image with amirstan plugin on actions

### ISSUE THREAD

**Issue by mmeendez8** [grimoire/amirstan_plugin#15]
Hi, this is somehow related with #14.

I am trying to build an image using the [build-push action](https://github.com/docker/build-push-action) for Nvidia Jetson architecture (linux/arm/v8).

I have a very simple dockerfile:

```docker
# syntax=docker/dockerfile:experimental

FROM nvcr.io/nvidia/deepstream-l4t:5.0.1-20.09-samples as build

WORKDIR / 

RUN --mount=type=cache,id=apt-build,target=/var/cache/apt \
    apt update && apt install -y \
        git \
        wget \
        cmake \
        g++ && \
    rm -rf /var/lib/apt/lists/*

RUN git clone --depth=1 --single-branch --branch patch-1 https://github.com/mmeendez8/amirstan_plugin.git && \ 
    cd /amirstan_plugin && \ 
    git submodule update --init --progress --depth=1 && \
    mkdir build && \
    cd build && \
    cmake .. -DWITH_DEEPSTREAM=true && \
    make -j10
```

Which works perfectly when it is build on Jetson... But this does not happen using docker multi-arch build on Github Actions or on my local computer.

I have set up a simple pipeline which builds the image using platforms: linux/arm64/v8 (see [here](https://github.com/mmeendez8/amirstan_plugin/blob/927f251818bccecc28dec68b75a7f864f0efcc4c/.github/workflows/deploy.yml))

The error is related with TensorRT, which is located in `/usr/include/aarch64-linux-gnu` but not found by the compiler...
```
#11 38.05 -- Found TensorRT headers at TENSORRT_INCLUDE_DIR-NOTFOUND
#11 38.06 -- Find TensorRT libs at TENSORRT_LIBRARY_INFER-NOTFOUND;TENSORRT_LIBRARY_PARSERS-NOTFOUND;TENSORRT_LIBRARY_INFER_PLUGIN-NOTFOUND
#11 38.07 -- Could NOT find TENSORRT (missing: TENSORRT_INCLUDE_DIR TENSORRT_LIBRARY) 
#11 38.07 ERRORCannot find TensorRT library.
```
I have tried to compile adding the path but with no success:
`cmake .. -DWITH_DEEPSTREAM=true -DTENSORRT_DIR=/usr/include/aarch64-linux-gnu`

Here is the complete log for the action: https://github.com/mmeendez8/amirstan_plugin/runs/2557972243?check_suite_focus=true

You can also test this on your local computer (if you have previously setup [qemu](https://www.stereolabs.com/docs/docker/building-arm-container-on-x86/)) using docker buildx with the following command:
`docker buildx build --platform linux/arm64 -f docker/Dockerfile -t amirstan_image_jetson:latest .`

**Comment by grimoire** [comment]
Sounds like a good way to add unit tests. Thanks, I will try this weekend.

**Comment by mmeendez8** [comment]
Yes, that's exactly what I am trying to do on my repo. I will be happy to help if you need it!

**Comment by mmeendez8** [comment]
I think it is definetely related with cmake version, it is not able to find tensort libraries with 3.10...

**Comment by grimoire** [comment]
Hi,
I do not find any TensorRT related heads or libs (NvInfer* or libnvinfer* ) in `nvcr.io/nvidia/deepstream-l4t:5.0.1-20.09-samples`

**Comment by mmeendez8** [comment]
Sorry for late reply, I found this in the Nvidia documentation about the Jetson docker images:

`The platform specific libraries and select device nodes for a particular device are mounted by the NVIDIA container runtime into the l4t-base container from the underlying host, thereby providing necessary dependencies for l4t applications to execute within the container. This approach enables the l4t-base container to be shared between various Jetson devices.`

**Comment by grimoire** [comment]
Did that mean the container will use TensorRT and other Dependent libraries from the host?

**Comment by mmeendez8** [comment]
I just found a solution. Can't believe how much time I spend on this... But finally hit me head with [this thread](https://forums.developer.nvidia.com/t/cmake-cannot-detect-cuda-dev-toolkit-inside-docker-when-running-in-jetson-agx-xaver-dev-kit/165519/8).

Summarizing, TensorRt is installed in the host system so you use it when you run you container by specifying `--runtime=nvidia` on your command. But I needed those libraries from the host on build time. The only way to solve this is by editing the docker daemon config file (/etc/docker/daemon.json) adding `"default-runtime": "nvidia"`. 

That should work but... I spend some more time figuring out why it was not working for me. It was related with Docker Buildkit which is not able to use the nvidia runtime on build stage. Once I disabled buildkit, everything was working as a charm!
